### PR TITLE
Infra Service resource and data source

### PR DESCRIPTION
**Pull request by mathewab**
- Added resource and data source implementation for infra/service

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -79,16 +79,22 @@ func (p *BloxOneProvider) Configure(ctx context.Context, req provider.ConfigureR
 func (p *BloxOneProvider) Resources(ctx context.Context) []func() resource.Resource {
 	return []func() resource.Resource{
 		ipam.NewIpSpaceResource,
+
```
Comment (venkat-iblox) [inline internal/provider/provider.go:82]: Do we need this space? Same with other spaces here and in Datasource.

Comment (mathewab) [reply internal/provider/provider.go:82]: Not needed, but thought it would be better for separating between packages

Comment (venkat-iblox) [reply internal/provider/provider.go:82]: Ok, then lets keep it as is

File: internal/service/infra_mgmt/api_services_data_source.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "",
+		Attributes: map[string]schema.Attribute{
+			"filters": schema.MapAttribute{
+				Description: "Filter are used to return a more specific list of results. Filters can be used to match resources by specific attributes, e.g. name. If you specify multiple filters, the results returned will have only resources that match all the specified filters.",
+				ElementType: types.StringType,
+				Optional:    true,
+			},
+			"tag_filters": schema.MapAttribute{
+				Description: "Tag Filters are used to filter by tags return a more specific list of results. If you specify multiple filters, the results returned will have only resources that match all the specified filters.",
```
Comment (venkat-iblox) [inline internal/service/infra_mgmt/api_services_data_source.go:57]: `Tag Filters return a list of results filtered by tags. If you specify multiple filters, the results returned will have only resources that match all the specified filters.`

Would this update make sense?

File: internal/service/infra_mgmt/model_infra_service.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+	},
+	"name": schema.StringAttribute{
+		Required:            true,
+		MarkdownDescription: "The name of the Service (unique).",
+	},
+	"pool_id": schema.StringAttribute{
+		Required:            true,
+		MarkdownDescription: "The resource identifier.",
+	},
+	"service_type": schema.StringAttribute{
```
Comment (venkat-iblox) [inline internal/service/infra_mgmt/model_infra_service.go:103]: The values that this field support, can we add these to the docs?

File: internal/service/infra_mgmt/api_services_data_source.go (hunk 2 above)
Comment (mathewab) [reply internal/service/infra_mgmt/api_services_data_source.go]: updated comment

File: internal/service/infra_mgmt/model_infra_service.go (hunk 3 above)
Comment (mathewab) [reply internal/service/infra_mgmt/model_infra_service.go:103]: added types in the description

Comment (venkat-iblox) [reply internal/service/infra_mgmt/model_infra_service.go:103]: Thank you

**Review by venkat-iblox — APPROVED** [review]

**Pull request merged**